stats/opentelemetry/client_metrics.go

```diff
@@ -63,6 +65,14 @@ func (csh *clientStatsHandler) unaryInterceptor(ctx context.Context, method stri
 	}
 	ctx = setCallInfo(ctx, ci)
 
+	if csh.o.MetricsOptions.pluginOption != nil {
+		md := csh.o.MetricsOptions.pluginOption.GetMetadata()
+		val := md.Get("x-envoy-peer-metadata")
```

Comment:
Range over all instead as discussed offline.

Reply:
Done. Kept the length check since this is a slice of values.